app/models/spree/subscription.rb

```diff
@@ -127,6 +127,17 @@ def send_prior_notification
       end
     end
 
+    def cancel_stripe_subscription
+      begin
+        stripe_subscription = Stripe::Subscription.retrieve(self.stripe_subscription_id)
+        stripe_subscription.cancel
+      rescue Stripe::StripeError => e
+        Rails.logger.error "Stripe error while canceling subscription: #{e.message}"
```

Comment:
if it fails how will admin know about the problem? system is not reporting anywhere